pkg/scheduler/scheduler_test.go

```diff
@@ -252,7 +252,292 @@ func TestLookupLatestPolicySnapshot(t *testing.T) {
 	}
 }
 
-// TestAddSchedulerCleanUpFinalizer tests the addSchedulerCleanUpFinalizer method.
+// TestAddSchedulerCleanUpFinalizerWithInterface tests the addSchedulerCleanUpFinalizer method with PlacementObj interface.
+func TestAddSchedulerCleanUpFinalizerWithInterface(t *testing.T) {
+	testCases := []struct {
+		name              string
+		placement         func() fleetv1beta1.PlacementObj
+		expectedFinalizer string
+	}{
+		{
+			name: "cluster-scoped placement should add CRP finalizer",
+			placement: func() fleetv1beta1.PlacementObj {
+				return &fleetv1beta1.ClusterResourcePlacement{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: crpName,
+					},
+				}
+			},
+			expectedFinalizer: fleetv1beta1.SchedulerCRPCleanupFinalizer,
+		},
+		{
+			name: "namespaced placement should add CRP finalizer",
+			placement: func() fleetv1beta1.PlacementObj {
+				return &fleetv1beta1.ResourcePlacement{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-rp",
+						Namespace: "test-namespace",
+					},
+				}
+			},
+			expectedFinalizer: fleetv1beta1.SchedulerCRPCleanupFinalizer,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			placement := tc.placement()
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme.Scheme).
+				WithObjects(placement).
+				Build()
+			s := &Scheduler{
+				client:         fakeClient,
+				uncachedReader: fakeClient,
+			}
+
+			ctx := context.Background()
+			if err := s.addSchedulerCleanUpFinalizer(ctx, placement); err != nil {
+				t.Fatalf("addSchedulerCleanUpFinalizer() = %v, want no error", err)
+			}
+
+			// Verify the finalizer was added
+			finalizers := placement.GetFinalizers()
+			found := false
+			for _, f := range finalizers {
+				if f == tc.expectedFinalizer {
+					found = true
+					break
+				}
+			}
+			if !found {
+				t.Errorf("Expected finalizer %s not found in %v", tc.expectedFinalizer, finalizers)
+			}
+		})
+	}
+}
+
+// TestCleanUpAllBindingsForWithInterface tests the cleanUpAllBindingsFor method with PlacementObj interface.
+func TestCleanUpAllBindingsForWithInterface(t *testing.T) {
+	now := metav1.Now()
+
+	testCases := []struct {
+		name              string
+		placement         func() fleetv1beta1.PlacementObj
+		bindings          []*fleetv1beta1.ClusterResourceBinding
+		expectedFinalizer string
```

Comment:
```suggestion
		wantFinalizer string
```
prefer "want" instead of expected